src/app/utils/firestore.ts

```diff
@@ -0,0 +1,62 @@
+import { Course, Question } from "../types/db";
+import { DocumentData, QueryDocumentSnapshot } from "firebase/firestore";
+
+export const courseConverter = {
+  toFirestore(course: Course): DocumentData {
+    return {
+      name: course.name,
+      location: course.location,
+      professors: course.professors,
+      students: course.students,
+      TAs: course.TAs,
+      onDuty: course.onDuty,
+      tags: course.tags,
+      // questions: course.questions,
+    };
+  },
+  fromFirestore(snapshot: QueryDocumentSnapshot, options: any): Course {
+    const data = snapshot.data(options);
+    if (!data) {
+      throw new Error("No data found in snapshot");
+    }
+    return {
+      name: data.name,
+      location: data.location,
+      professors: data.professors,
+      students: data.students,
+      TAs: data.TAs,
+      onDuty: data.onDuty,
+      tags: data.tags,
+      // questions: data.questions,
+    };
+  },
+};
+
+export const questionConverter = {
+  toFirestore(question: Question): DocumentData {
+    return {
+      title: question.title,
+      description: question.description,
+      public: question.public,
+      state: question.state,
+      timestamp: question.timestamp,
```

Comment:
I think timestamp should be generated on server side. If we generate it on client, it's possible to modify the clock

Reply:
https://stackoverflow.com/questions/69519447/how-to-get-server-timestamp-from-firebase-v9